espnet/nets/pytorch_backend/nets_utils.py

```diff
@@ -60,10 +60,56 @@ def pad_list(xs, pad_value):
 
     return pad
 
-
-def make_pad_mask(lengths, xs=None, length_dim=-1, maxlen=None):
+def make_pad_mask(lengths, maxlen=None):
     """Make mask tensor containing indices of padded part.
 
+    Args:
+        lengths (LongTensor or List): Batch of lengths (B,).
+
+    Returns:
+        BoolTensor: Mask tensor containing indices of padded part.
+
+    Examples:
+        >>> lengths = [5, 3, 2]
+        >>> make_pad_mask(lengths)
+        masks = [[0, 0, 0, 0 ,0],
+                 [0, 0, 0, 1, 1],
+                 [0, 0, 1, 1, 1]]
+    """
+    lengths = lengths.long()
```

Comment:
When `type(lengths)` is `List`, this line will cause an error.
If we support both `torch.Tensor` and `List`, then we need to add if sentence like

```python
if isinstance(lengths, List):
    lengths = torch.Tensor(lengths)
```

Reply:
Yeah, I also thought about this, but wasn't sure exactly in which cases we are passing in tensors vs lists. @Masao-Someki do you know if the ONNX export is able to properly build the computational graph in the case that it is a Python list? I was assuming that it might not work, so was thinking it might be better to change everything into tensors.

Reply:
@kenzheng99
I think torch can export the model properly. I'm not sure if any function call uses this function with `List` as input, but I think limiting the input type is better for readability.